frontends/infinite-corridor/src/pages/field-details/EditFieldAppearanceForm.test.tsx

```diff
@@ -6,20 +6,26 @@ import {
   user
 } from "../../test-utils"
 import * as factory from "../../api/fields/factories"
-import { urls } from "../../api/fields"
+import { FieldChannel, urls } from "../../api/fields"
+import { urls as widgetUrls } from "../../api/widgets"
 import { waitFor } from "@testing-library/react"
 import { makeFieldViewPath } from "../urls"
+import { makeWidgetListResponse } from "ol-widgets"
 
-describe("EditFieldAppearanceForm", () => {
-  let field
-
-  beforeEach(() => {
-    field = factory.makeField({ is_moderator: true })
-    setMockResponse.get(urls.fieldDetails(field.name), field)
-    setMockResponse.get(urls.userLists(), [field])
-  })
+const setupApis = (fieldOverrides?: Partial<FieldChannel>) => {
+  const field = factory.makeField({ is_moderator: true, ...fieldOverrides })
+  setMockResponse.get(urls.fieldDetails(field.name), field)
+  setMockResponse.get(urls.userLists(), [field])
+  setMockResponse.get(
+    widgetUrls.widgetList(field.widget_list),
+    makeWidgetListResponse({}, { count: 0 })
+  )
+  return field
+}
 
+describe("EditFieldAppearanceForm", () => {
   it("Displays the field title, appearance inputs with current field values", async () => {
+    const field = setupApis()
```

Comment:
Mostly: you can pass parameters to `setupApis`, whereas you can't pass parameters to beforeEach hooks. So, below I can call it as:
```ts
const field = setupApis({featured_list: null, lists: []})
```
which I wanted to do so I didn't have to set up some mock API calls.¹

---
¹"So I don't have to set up mock API calls" is maybe a separate issue that's worth thinking about. I like the tests that use `renderTestApp` because they render the whole app which is more realistic. But it is a bit of a pain to things like "Clicking a link on PageX sends you to PageY", because you end up having to provide mock API responses for both pages X and Y... One could say "So don't write tests that way, just mock PageY". I tend to think it would be better to just make the API mocking easier... e.g., via [msw](https://github.com/mswjs/msw). _Nathan pointed me toward msw a while ago—we don't use it anywhere, but I've been trying it out in a side project._